fecgov/openFEC
# Recalculate most_recent flag of filings by using v_sum table

## Summary (required)

- Resolves #[_3946_](https://github.com/fecgov/openFEC/issues/3946)

Currently most_recent flag is obtained from ofec_filings_amendments_all_mv which collects data by two ways: 
1) using the relationship between file_num and previous_file_num 
2) partition over cmte_id,form_tp order by file_num

Because previous_file_num is not updated sometimes, especially for paper filing, this PR change the logic to use v_sum table to check if a filling is the most recent. And for F1, F1M, and F2 which are not in v_sum, calculation is partitioned over cmte_id,form_tp order by receipt_date, begin_image_number

## How to test the changes locally

1)   Download the branch to local, run flyway to make sure migration run smoothly
2)  To test endpoint:  
      endpoint:  /filings/ 
      SQLA_CONN: DEV
      webservices/common/models/filings.py:   
                   class Filings(FecFileNumberMixin, CsvMixin, db.Model):
                   __tablename__ = 'ofec_filings_all_mv_tmp_hc'
      
     2.1)  committee_id=C00334029, rpt_yr=2018, 'current version' should be seen for quarterly reports
 http://127.0.0.1:5000/v1/filings/?page=1&sort_hide_null=false&sort=-receipt_date&per_page=20&sort_null_only=false&sort_nulls_last=false&committee_id=C00334029&report_year=2018
   2.2)  committee_id=C00665554, rpt_yr=2018, rpt_tp=Q2, Current version is '1350957' 
http://127.0.0.1:5000/v1/filings/?page=1&sort_hide_null=false&sort=-receipt_date&per_page=20&report_type=Q2&sort_null_only=false&sort_nulls_last=false&committee_id=C00665554&report_year=2018


Current Production:

<img width="900" alt="Screen Shot 2019-09-30 at 4 04 23 PM" src="https://user-images.githubusercontent.com/24396726/65912386-1b10cd00-e39c-11e9-9159-4415ccaa2078.png">
====================

Local CMS after:

<img width="900" alt="Screen Shot 2019-09-30 at 4 06 55 PM" src="https://user-images.githubusercontent.com/24396726/65912652-88246280-e39c-11e9-9d66-1af52bdb5d0a.png">
